### PR TITLE
Remove defaulting FSGroup

### PR DESCRIPTION
**Pull request by hossainemruz**
Signed-off-by: hossainemruz <emruz@appscode.com>

### CODE REVIEW

File: pkg/controller/backup_configuration.go [hunk 1]
```diff
@@ -255,22 +255,39 @@ func (c *StashController) EnsureBackupTriggeringCronJob(invoker apis.Invoker) er
...
-				},
-			})
+			in.Spec.JobTemplate.Spec.Template.Spec.Containers, container)
 		in.Spec.JobTemplate.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
 		in.Spec.JobTemplate.Spec.Template.Spec.ServiceAccountName = serviceAccountName
-		// insert default pod level security context
-		in.Spec.JobTemplate.Spec.Template.Spec.SecurityContext = util.UpsertDefaultPodSecurityContext(in.Spec.JobTemplate.Spec.Template.Spec.SecurityContext)
+
+		// only apply the pod level runtime settings that make sense for the CronJob
+		if invoker.RuntimeSettings.Pod != nil {
```
Comment (tamalsaha) [inline pkg/controller/backup_configuration.go]: Why not apply the whole thing?

```
	// Pass pod RuntimeSettings from RestoreSession
	if rs.Spec.RuntimeSettings.Pod != nil {
		jobTemplate.Spec = ofst_util.ApplyPodRuntimeSettings(jobTemplate.Spec, *rs.Spec.RuntimeSettings.Pod)
	}
```

File: pkg/controller/backup_configuration.go [hunk 2]
```diff
@@ -255,22 +255,39 @@ func (c *StashController) EnsureBackupTriggeringCronJob(invoker apis.Invoker) er
...
+			ImagePullPolicy: core.PullIfNotPresent,
+			Image:           image.ToContainerImage(),
+			Args: []string{
+				"create-backupsession",
+				fmt.Sprintf("--invoker-name=%s", invoker.OwnerRef.Name),
+				fmt.Sprintf("--invoker-type=%s", invoker.OwnerRef.Kind),
+			},
+		}
+		// only apply the container level runtime settings that make sense for the CronJob
+		if invoker.RuntimeSettings.Container != nil {
```
Comment (tamalsaha) [inline pkg/controller/backup_configuration.go]: Why not apply all the container runtime settings?

Comment (hossainemruz) [reply pkg/controller/backup_configuration.go]: We can't. If user specify `liveness` and `readiness` probe, then the CronJob might fail.

File: pkg/controller/backup_configuration.go (hunk 1 above)
Comment (hossainemruz) [reply pkg/controller/backup_configuration.go]: Does others setting make sense in respect of CronJob?

**Pull request merged**